# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/nsAccessibilityService.h
# Commit: 2ba84268f8ac
# Full Hash: 2ba84268f8ac2baf36477e62bc03ade97966d865
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2025-09-30 17:19 +0000
# Regressor Bug: 1987945
# File Overlap Count: 1
# Overlapping Files: accessible/tests/browser/relations/browser_anchor_positioning.js
# Description:
#   Bug 1987945 - P3: Support details relation in anchored targets in remote accessibles. r=Jamie,layout-anchor-positioning-reviewers,layout-reviewers,emilio,firefox-style-system-reviewers
# ==============================================================================

diff -r 7aaaa4514127 -r 2ba84268f8ac accessible/base/nsAccessibilityService.h
--- a/accessible/base/nsAccessibilityService.h	Tue Sep 30 17:19:31 2025 +0000
+++ b/accessible/base/nsAccessibilityService.h	Tue Sep 30 17:19:31 2025 +0000
@@ -267,6 +267,23 @@
                                    int32_t aAppUnitsPerDevPixel);
 
   /**
+   * Notify accessibility that an anchor positioned frame is
+   * about to be removed. This gives us a chance to update cached relations
+   * before the reflow where we will lose references to the anchor and won't be
+   * able to refresh its accessible's cache.
+   */
+  void NotifyAnchorPositionedRemoved(mozilla::PresShell* aPresShell,
+                                     nsIFrame* aFrame);
+
+  /**
+   * Notify accessibility that an anchor frame is about to be removed. This
+   * gives us a chance to update cached relations before the reflow where the
+   * anchor will be lost and we won't be able to refresh the accessible cache of
+   * prior relations.
+   */
+  void NotifyAnchorRemoved(mozilla::PresShell* aPresShell, nsIFrame* aFrame);
+
+  /**
    * Notify accessibility that an element explicitly set for an attribute is
    * about to change. See dom::Element::ExplicitlySetAttrElement.
    */